src/Tizen.Diagnostics/Tizen.Diagnostics/Diagnostics.cs

```diff
@@ -0,0 +1,294 @@
+
+            var ret = NativeApi.RequestClientData(clientId, parameters, parameters.Length, out IntPtr data);
+            if (ret != NativeApi.DiagnosticsError.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return new DiagnosticsData(data);
+        }
+
+        private static void SetNotificationCallback()
+        {
+            _notificationCallback = (IntPtr ctx, IntPtr data) =>
+            {
+                _reportReceived?.Invoke(null, new ReportReceivedEventArgs(ctx));
+            };
+
+            var ret = NativeApi.SetNotificationCb(_notificationCallback, IntPtr.Zero);
+            if (ret != NativeApi.DiagnosticsError.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        private static void UnsetNotificationCallback()
+        {
+            NativeApi.UnsetNotificationCb();
+        }
+    }
+
+    /// <summary>
+    /// The DiagnosticsContext class is used for getting specific type of the report from the context.
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public class DiagnosticsContext : IDisposable
+    {
+        private IntPtr _handle = IntPtr.Zero;
+        private bool _disposed = false;
+
+        /// <summary>
+        /// The DiagnosticsContext class constructor.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="handle">Diagnostics context received in the ReportReceived event.</param>
+        internal DiagnosticsContext(IntPtr handle)
+        {
+            _handle = handle;
+
+            var ret = NativeApi.GetType(handle, out ContextType contextType);
+            if (ret != NativeApi.DiagnosticsError.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            Type = contextType;
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the DiagnosticsContext class.
+        /// </summary>
+        ~DiagnosticsContext()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Gets report of specific type from the context.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="report">Report type to retrieve from the context.</param>
+        /// <returns>
+        /// The instance of DiagnosticsData.
+        /// </returns>
+        /// <remarks>
+        /// This function is permitted only to an app signed by platform level certificates.
+        /// </remarks>
+        public DiagnosticsData GetReport(ReportType report)
+        {
+            var ret = NativeApi.GetReport(_handle, report, out IntPtr data);
+            if (ret != NativeApi.DiagnosticsError.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return new DiagnosticsData(data);
+        }
+
+        /// <summary>
+        /// Stores context type.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public ContextType Type { get; internal set; }
```

Comment:
`Type` is ambiguous with `Object.GetType`
```suggestion
        public ContextType ContextType { get; internal set; }
```